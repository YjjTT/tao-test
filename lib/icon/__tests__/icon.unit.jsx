/*
 * @Author: yjjtt
 * @Date: 2019-09-11 13:19:41
 * @Last Modified by: yjjtt
 * @Last Modified time: 2019-09-16 17:06:27
 */
import React from "react";
import renderer from "react-test-renderer";
import Icon from "../icon";
import { mount } from "enzyme";

describe("icon", () => {
  it("render success 是个 svg", () => {
    const json = renderer.create(<Icon />).toJSON();
    expect(json).toMatchSnapshot();
  });

  it("onClick", () => {
    const fn = jest.fn()
    const component = mount(<Icon name="alipay" onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
