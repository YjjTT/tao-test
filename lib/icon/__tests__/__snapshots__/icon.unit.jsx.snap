// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render success 是个 svg 1`] = `
<svg
  className="tui-icon"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
